### PR TITLE
ROX-22019: Tests to cover protocompat time functions with errors

### PR DESCRIPTION
**Pull request by mtodor**
## Description

This PR is adding unit tests (or sections) that will cover execution paths that return an error.

This is important for protobuf V2 migration because `gogo` proto conversion functions do validation, but protobuf V2 (or V1) framework does not do that, and validation has to be called manually. To avoid wrong implementation of compatibility functions with protobuf V2 we should cover paths where errors can be returned.

The intention of added tests is not to cover validation implemented in a specific proto framework, but to cover that compatibility functions can return an error. We had an issue during the protobuf V2 switch that changes in the compatibility layer did not ever return an error, and that caused unexpected behavior in caller functions.

Implementation of validations are different in different frameworks. One version allows 10000-01-01 00:00:00 and another fails for that. We do not need to cover all edge cases.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

### Here I tell how I validated my change

- only CI unit test runs

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/protocompat/time_test.go [hunk 1]
```diff
@@ -115,6 +115,27 @@ func TestConvertTimeToTimestampOrNil(t *testing.T) {
...
+	protoTime := &types.Timestamp{
+		Seconds: 2345678901,
+		Nanos:   234567891,
+	}
+	gqlTime, err = ConvertTimestampToGraphqlTimeOrError(protoTime)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(2345678901234567891), gqlTime.UnixNano())
+
+	invalidProtoTime := &types.Timestamp{
+		Seconds: 253402300801,
```
Comment (rhybrillou) [inline pkg/protocompat/time_test.go:132]: We should probably use constants with appropriate comments to represent valid `Seconds` and `Nanos` values within `Timestamp` and `Duration` objects.

Values for `Timestamp` are (https://github.com/gogo/protobuf/blob/f67b8970b736e53dbd7d0a27146c8f1ac52f74e5/types/timestamp.go#L43-L48) :
minValidSeconds: -62135596800 // time.Date(1,1,1,0,0,0,0, time.UTC).Unix()
maxValidSeconds: 253402300800 // time.Date(10000,1,1,0,0,0,0,time.UTC).Unix()

The given values for Duration seem to not match what provides the actual outcome of the validation function.
https://github.com/gogo/protobuf/blob/f67b8970b736e53dbd7d0a27146c8f1ac52f74e5/types/duration.go#L44-L47
The expected value would be 10000 years. With the overflow check taken into account, the values I found as being the boundaries are:
maxDurationSeconds = 9223372036
minDurationSeconds = -9223372036

File: pkg/protocompat/time_test.go [hunk 2]
```diff
@@ -208,6 +229,13 @@ func TestDurationFromProto(t *testing.T) {
 	timeDuration, err := DurationFromProto(protoDuration)
 	assert.NoError(t, err)
 	assert.Equal(t, expectedDuration, timeDuration)
+
+	protoInvalidDuration := &types.Duration{
+		Seconds: 253402300801,
```
Comment (rhybrillou) [inline pkg/protocompat/time_test.go:234]: Where does this value come from ? It looks like it is the number of seconds from the Unix Epoch to the first invalid timestamp second.

Comment (rhybrillou) [reply pkg/protocompat/time_test.go:234]: Do we want to test the various Duration validation cases ?
- Seconds below minSeconds (corrected for overflow)
- Seconds above maxSeconds (corrected for overflow)
- Nanos below -999999999
- Nanos above 999999999
- Sign difference between seconds and nanos (in both directions)
- Resulting duration is not of the same sign as input duration Nanos

Comment (mtodor) [reply pkg/protocompat/time_test.go]: I have added additional information in the description. The intention is not to cover all validation edge cases but to ensure that an error will be returned for an invalid value.

File: pkg/protocompat/time_test.go (hunk 1 above)
Comment (mtodor) [reply pkg/protocompat/time_test.go]: I have changed tests. I don't want to use edge values.

One framework uses `253402300800` (gogo proto).
Another uses `253402300799`(google proto).

I'm using "common" sense validation cases. And I have added comments with explanations. (no constants)

**Review by rhybrillou — APPROVED** [review]

**Pull request merged**